packages-exp/auth-exp/src/core/auth/emulator.ts

```diff
@@ -58,22 +58,60 @@ export function useAuthEmulator(
     AuthErrorCode.INVALID_EMULATOR_SCHEME
   );
 
-  const parsedUrl = new URL(url);
   const disableWarnings = !!options?.disableWarnings;
 
-  // Store the normalized URL whose path is always nonempty (i.e. containing at least a single '/').
-  authInternal.config.emulator = { url: parsedUrl.toString() };
+  const protocol = extractProtocol(url);
+  const { host, port } = extractHostAndPort(url);
+  const portStr = port === null ? '' : `:${port}`;
+
+  // Always replace path with "/" (even if input url had no path at all, or had a different one).
+  authInternal.config.emulator = { url: `${protocol}//${host}${portStr}/` };
   authInternal.settings.appVerificationDisabledForTesting = true;
   authInternal.emulatorConfig = Object.freeze({
-    host: parsedUrl.hostname,
-    port: parsedUrl.port ? Number(parsedUrl.port) : null,
-    protocol: parsedUrl.protocol.replace(':', ''),
+    host,
+    port,
+    protocol: protocol.replace(':', ''),
     options: Object.freeze({ disableWarnings })
   });
 
   emitEmulatorWarning(disableWarnings);
 }
 
+function extractProtocol(url: string): string {
+  const protocolEnd = url.indexOf(':');
+  return protocolEnd < 0 ? '' : url.substr(0, protocolEnd + 1);
+}
+
+function extractHostAndPort(
+  url: string
+): { host: string; port: number | null } {
+  const protocol = extractProtocol(url);
```

Comment:
Should we always try to use URL() first in the happy path? That way this code is only being exercised on a few platforms

Reply:
Impl standpoint: `new URL(input)` works fine in React Native, it just throws on `urlInstance.hostname` the getter. This may change at any time and some methods are stubs or half-baked implementation and won't throw. This makes it annoying to feature-detect and I'd rather not work with such fragile assumptions. It will lead to hard to debug issues with combinations of React Native versions and TS SDK versions (`O(M*N)`).

Testing standpoint: I'd rather just have one code path that is not platform dependent and always tested rather than two. Our implementation is not a proper polyfill and let's not pretend it is. If something breaks here we can just fix it for all platforms.

Reply:
Fair enough. I'm fine with that since this is just for enabling the emulator, not in any code that is meant to reach production